python/pyspark/ml/param/__init__.py

```diff
@@ -280,7 +281,9 @@ def explainParam(self, param):
         else:
             values.append("undefined")
         valueStr = "(" + ", ".join(values) + ")"
-        return "%s: %s %s" % (param.name, param.doc, valueStr)
+        # Remove the default value from the doc string
+        docStr = re.sub(r'\(Default [\w\-\s]+?\)\Z', '', param.doc)
+        return "%s: %s %s" % (param.name, docStr, valueStr)
```

Comment:
Further more, I don't think using regex is an elegant way to remove the default value in users specified docs. Users can use various words to express ```default``` that we can not filter them completely.